### PR TITLE
[sui-proxy/static keys] allow intake of static pub keys

### PR DESCRIPTION
**Pull request by suiwombat**
## Description 

we may want to provide static pub keys to accept when receiving metrics from sui-node; this pr allows us to do that.
this is also useful if we need to allow any general node to send us data and not be in the consensus committee.  

there is a breaking change for sui-proxy, the config format have changed to support the idea of dynamic and static keys, but this is minor and will be adjusted along with the static key config.

## Test Plan 
local testing

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-proxy/src/ip.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    let allowed_private: IpNetwork = "169.254.0.0/16".parse().unwrap();
+    if allowed_private.contains(IpAddr::V4(addr)) {
+        // intentional
+        return false;
+    }
+    addr.is_private()
+}
+
+/// is_private_v6 and the funcs below are based on an unstable const fn in core. yoinked it.
+/// taken from https://doc.rust-lang.org/nightly/src/core/net/ip_addr.rs.html#1501-1531
```
Comment (johnjmartin) [inline crates/sui-proxy/src/ip.rs:35]: I doubt this code this url is pointing at is going to remain constant, doesn't the code in nightly change frequently?

File: crates/sui-proxy/src/ip.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    match addr {
+        IpAddr::V4(a) => is_private_v4(a),
+        IpAddr::V6(a) => is_private_v6(a),
+    }
+}
+
+/// is_private_v4 will say just that, is it private? we ignore 169.254.0.0/16 in this consideration
+fn is_private_v4(addr: Ipv4Addr) -> bool {
+    // special case we will allow
+    let allowed_private: IpNetwork = "169.254.0.0/16".parse().unwrap();
```
Comment (johnjmartin) [inline crates/sui-proxy/src/ip.rs:26]: what's the significance of this address?

File: crates/sui-tls/src/verifier.rs [hunk 3]
```diff
@@ -189,7 +189,7 @@ fn pki_error(error: webpki::Error) -> rustls::Error {
     }
 }
 
-pub(crate) fn public_key_from_certificate(
```
Comment (johnjmartin) [inline crates/sui-tls/src/verifier.rs:192]: what's the reason for this change?

**Review by johnjmartin — APPROVED** [review]
🚀

File: crates/sui-proxy/src/ip.rs (hunk 2 above)
Comment (suiwombat) [reply crates/sui-proxy/src/ip.rs:26]: its not routable.  i tend to view 169.254.0.0/16 addresses as special local network cases.  for lack of any better choice, i picked this range since the internal code requires a multiaddr type, which in turn requires an ip address of some kind.  there is plenty of space in a /16 and it's a special case. it seemed to fit the need.

File: crates/sui-proxy/src/ip.rs (hunk 1 above)
Comment (suiwombat) [reply crates/sui-proxy/src/ip.rs]: updated. good point.
